rasa/core/brokers/kafka.py

```diff
@@ -198,7 +200,16 @@ def _publish(self, event: Dict[Text, Any]) -> None:
         logger.debug(
             f"Calling kafka send({self.topic}, value={event}, key={partition_key!s})"
         )
-        self.producer.send(self.topic, value=event, key=partition_key)
+
+        headers = [("RASA_ENVIRONMENT", bytes(self.rasa_environment, encoding=DEFAULT_ENCODING))]
```

Comment:
Can we have a unit test for this?

Reply:
Trying to figure out how to check this because it's not a property like it is for Pika - it only shows up in the headers when the event is sent. Is it necessary to mock a consumer to check what is sent, or do you think it can be mocked higher up?

Reply:
If it requires everything to be mocked then probably not worth it. We really need some integration tests using an actual kafka.

Reply:
Gotcha. So no unit tests on this PR then? All we can assert without mocks is that the environment var is fetched but that seems a little ¯\_(ツ)_/¯